app/controllers/questions_controller.rb

```diff
@@ -1,9 +1,13 @@
 class QuestionsController < ApplicationController
+  before_action :set_test, only: %i[ show edit update destroy ]
   before_action :set_question, only: %i[ show edit update destroy ]
 
+  QUESTIONS_PER_PAGE = 3
   # GET /questions or /questions.json
   def index
-    @questions = Question.all
+    #@questions = Question.all
+    @page = params.fetch(:page, 0).to_i
```

Comment:
Let's use `pagy` gem for pagination
https://ddnexus.github.io/pagy/extras/headers#gsc.tab=0